Remove test dependency from package:flutter

abarth opened the pull request:
The mojo_sdk has cleaned up its dependencies, so we no longer need to depend on
package:test from package:flutter.

abarth commented:
@Hixie

Hixie commented on packages/flutter/pubspec.yaml:
```diff
@@ -7,14 +7,12 @@ homepage: http://flutter.io
-  # See the comment in flutter_tools' pubspec.yaml. We have to pin it
-  # here also because sky_services depends on mojo_sdk which depends
-  # on test.
-  test: 0.12.6+1
```
also fix the comment in flutter_tools' pubspec.yaml

abarth committed:
Remove test dependency from package:flutter

The mojo_sdk has cleaned up its dependencies, so we no longer need to depend on
package:test from package:flutter.

Hixie commented:
lgtm
